tests/tom-swifty-test/SwiftReflector/ProtocolTests.cs

```diff
@@ -104,6 +104,8 @@ void WrapSingleSubscriptGetOnly (string type, string csType, string csReplacemen
 			TestRunning.TestAndExecute (swiftCode, callingCode, expected, testName : $"WrapSingleSubscriptGetOnly{type}", otherClass : overCS);
 		}
 
+#if _MAC_TS_TEST_
```

Comment:
`_MAC_TS_TEST_ ` doesn't make sense here.  It should only be used in generated code, not in an actual test. I want to be clear that my tone here might sound curt, but I want to be specific. It's NBD here, but it doesn't do what you think it does.

TL;DR - I want to make a test that runs only on a specific platform. How do I do that? There's an argument for that. When you call `TestAndExecute` there is an optional argument, `platform` (IIRC) which you can use to specify "I want this to run on....". If you don't specify it, the assumption is that it will run on all platforms upon which we run tests (currently macOS and iOS, but in the future...).

What is `_MAC_TS_TEST_` for then? A lot of it comes down to "thanks, swift, you're the best!". There are some tests that are written that do printing from swift (DON'T DO THIS). This turns out to be a bad idea because the (1) swift output and mono output are buffered differently and the ordering can be surprising and not always consistent. (2) I've yet to succeed in getting swift output redirected on iOS transparent to the swift code. (3) sometimes the set up and tear down for the actual test has things that only make sense on macOS.
Because of this, test generation code knows that it needs to make blocks.
For a typical unit test, the process is:
1. write swift code and compile it
2. creating bindings for the swift code and compile the swift parts
3. write C# code to invoke the binding in some meaningful way and generate output
4. compile the C# code and execute it
5. collect the output and see if it matches the expected `Assert.AreEqual (output, expected);`

If you look in `CaptureSwiftOutputPostlude` in `TestRunning.cs` you can see where `_MAC_TS_TEST_` gets injected into the code. If you look in `Compiler.cs` you can see where `_MAC_TS_TEST_` gets set as a compiler option (`-d _MAC_TS_TEST`).

If you put this around the actual `[Test]` block, what you're doing is commenting out the test so it never gets run.

Lessons learned for the future:
1. Testing binding is hard and you should think of this whole thing (gestures vaguely) as a dancing bear. It's not that it dances well that's impressive, it's that it dances at all.
2. Avoid writing unit tests that depend on output in swift. We can do it, but it's a serious pain and every time you run a unit test with swift output, a pixie gets set on fire.
3. If you can come up with a better way to unit test this (runs faster, easier to debug, etc), I am all ears. This is not defensive - I would seriously love an improvement to this process. I have a few ideas, but they're all trade-offs (faster tests, but harder to debug or faster tests but unrealistic testing scenario).